### PR TITLE
Resolves segfault issue on ClearHistory() function call

### PR DESCRIPTION
**Pull request by jreyes512**
Edited RemoveDisplay(item) function to delete the map value that was asked to be removed, rather than replacing it with an unititalized pointer.

Presence of empty pointer resulted in segfault during ClearHistory() function call.
Failsafe added to Clear History to ignore empty mapped values.

Resolves segfault occuring when any plugin had been removed during the active session prior to calling ClearHistory.

Second commit removes unnecessary files.

### CODE REVIEW

File: mapviz/src/map_canvas.cpp [hunk 1]
```diff
@@ -464,8 +464,12 @@ void MapCanvas::AddPlugin(MapvizPluginPtr plugin, int order)
 
 void MapCanvas::RemovePlugin(MapvizPluginPtr plugin)
 {
+  
+  //plugin->ClearHistory();
```
Comment (pjreed) [inline mapviz/src/map_canvas.cpp]: I'd go ahead and remove this comment if it doesn't need to be here.

File: mapviz/src/mapviz.cpp [hunk 2]
```diff
@@ -921,9 +921,26 @@ void Mapviz::SaveConfig()
 
 void Mapviz::ClearHistory()
 {
+  ROS_INFO("made it to loop");
```
Comment (pjreed) [inline mapviz/src/mapviz.cpp]: I'm guessing this was just added for debugging purposes; it should either be `ROS_DEBUG` or removed.
```suggestion
  ROS_DEBUG("Mapviz::ClearHistory()");
```

File: mapviz/src/mapviz.cpp [hunk 3]
```diff
@@ -1441,7 +1458,7 @@ void Mapviz::Screenshot()
   }
   else
   {
-    ROS_ERROR("Failed to take screenshot.");
+    ROS_ERROR("Failed to take/opt/ros/kinetic/lib/librqt_mapviz.so screenshot.");
```
Comment (pjreed) [inline mapviz/src/mapviz.cpp]: This looks accidental, did this just get accidentally pasted here?
```suggestion
    ROS_ERROR("Failed to take screenshot.");
```

File: mapviz_plugins/src/laserscan_plugin.cpp [hunk 4]
```diff
@@ -161,6 +161,7 @@ namespace mapviz_plugins
 
   void LaserScanPlugin::ClearHistory()
   {
+    ROS_INFO("LaserScan::ClearHistory(test)");
```
Comment (pjreed) [inline mapviz_plugins/src/laserscan_plugin.cpp]:
```suggestion
    ROS_DEBUG("LaserScan::ClearHistory()");
```

File: mapviz/src/mapviz.cpp [hunk 5]
```diff
@@ -1474,6 +1491,7 @@ void Mapviz::RemoveDisplay(QListWidgetItem* item)
   {
     canvas_->RemovePlugin(plugins_[item]);
     plugins_[item] = MapvizPluginPtr();
+    plugins_.erase(item);
```
Comment (pjreed) [inline mapviz/src/mapviz.cpp]: Does adding an empty pointer do anything if it's being erased immediately afterward?

File: mapviz_plugins/src/odometry_plugin.cpp [hunk 6]
```diff
@@ -91,12 +91,21 @@ namespace mapviz_plugins
                      SLOT(CovariancedToggled(bool)));
     QObject::connect(ui_.buttonResetBuffer, SIGNAL(pressed()), this,
                      SLOT(ClearPoints()));
+    //QObject::connect(ui_.clear, SIGNAL(clicked()), this, SLOT(ClearHistory()));
```
Comment (pjreed) [inline mapviz_plugins/src/odometry_plugin.cpp]: Should this be commented out?

File: mapviz_plugins/src/odometry_plugin.cpp [hunk 7]
```diff
@@ -91,12 +91,21 @@ namespace mapviz_plugins
...
+
   }
 
   OdometryPlugin::~OdometryPlugin()
   {
   }
 
+  void OdometryPlugin::ClearHistory()
+  {
+    ROS_INFO("OdometryPlugin::ClearHistory()");
```
Comment (pjreed) [inline mapviz_plugins/src/odometry_plugin.cpp]:
```suggestion
    ROS_DEBUG("OdometryPlugin::ClearHistory()");
```

File: mapviz_plugins/src/point_drawing_plugin.cpp [hunk 8]
```diff
@@ -65,6 +65,7 @@ namespace mapviz_plugins
 
   void PointDrawingPlugin::ClearHistory()
   {
+    ROS_INFO("PointDrawingPlugin::ClearHistory(test)");
```
Comment (pjreed) [inline mapviz_plugins/src/point_drawing_plugin.cpp]:
```suggestion
    ROS_DEBUG("PointDrawingPlugin::ClearHistory()");
```

File: mapviz_plugins/src/pointcloud2_plugin.cpp [hunk 9]
```diff
@@ -176,6 +176,7 @@ namespace mapviz_plugins
 
   void PointCloud2Plugin::ClearHistory()
   {
+    ROS_INFO("PointCloud2Plugin::ClearHistory(test)");
```
Comment (pjreed) [inline mapviz_plugins/src/pointcloud2_plugin.cpp]:
```suggestion
    ROS_DEBUG("PointCloud2Plugin::ClearHistory()");
```

File: mapviz/src/mapviz.cpp (hunk 3 above)
Comment (jreyes512) [reply mapviz/src/mapviz.cpp]: Yeah this was accidental.  Oops

File: mapviz/src/mapviz.cpp (hunk 2 above)
Comment (jreyes512) [reply mapviz/src/mapviz.cpp]: Correct, for debugging.  Will remove.

File: mapviz_plugins/src/point_drawing_plugin.cpp (hunk 8 above)
Comment (pjreed) [reply mapviz_plugins/src/point_drawing_plugin.cpp]: I'm not sure you meant to resolve this one without changing it...

File: mapviz/src/map_canvas.cpp [hunk 10]
```diff
@@ -464,8 +464,12 @@ void MapCanvas::AddPlugin(MapvizPluginPtr plugin, int order)
 
 void MapCanvas::RemovePlugin(MapvizPluginPtr plugin)
 {
+
```
Comment (pjreed) [inline mapviz/src/map_canvas.cpp]:
```suggestion
  plugin->Shutdown();
  plugins_.remove(plugin);
```

File: mapviz/src/mapviz.cpp [hunk 11]
```diff
@@ -921,9 +921,27 @@ void Mapviz::SaveConfig()
...
+    {
+      // May not be necessary now, but good to have as a failsafe / debugging tool.  
+      // Attempting to clear empty map elements previously led to segfault
+      
+      // trying to acces which plugins crash also result in a segfault.
+      //QString qs = plugin.first->text(); 
+      //std::string s = qs.toUtf8().constData();
+      ROS_INFO("Clear Failed.");
+    }
+
```
Comment (pjreed) [inline mapviz/src/mapviz.cpp]: Let's go ahead and remove the excess whitespace here, too.
```suggestion
  ROS_DEBUG("Mapviz::ClearHistory()");
  for (auto& plugin: plugins_)
  {
    plugin.second->ClearHistory();
    if (plugin.second)
    {
      plugin.second->ClearHistory();
    }
    else
    {
      // May not be necessary now, but good to have as a failsafe / debugging tool.  
      // Attempting to clear empty map elements previously led to segfault
      // trying to access which plugins crash also result in a segfault.
      ROS_INFO("Clear Failed.");
    }
```

File: mapviz_plugins/src/odometry_plugin.cpp [hunk 12]
```diff
@@ -91,12 +91,21 @@ namespace mapviz_plugins
                      SLOT(CovariancedToggled(bool)));
     QObject::connect(ui_.buttonResetBuffer, SIGNAL(pressed()), this,
                      SLOT(ClearPoints()));
+    
+
```
Comment (pjreed) [inline mapviz_plugins/src/odometry_plugin.cpp]: More unnecessary empty space.

File: mapviz_plugins/src/odometry_plugin.cpp [hunk 13]
```diff
@@ -91,12 +91,21 @@ namespace mapviz_plugins
...
   OdometryPlugin::~OdometryPlugin()
   {
   }
 
+  void OdometryPlugin::ClearHistory()
+  {
+    ROS_DEBUG("OdometryPlugin::ClearHistory()");
+    ClearPoints();
+  }
+
```
Comment (pjreed) [inline mapviz_plugins/src/odometry_plugin.cpp]: More unnecessary empty space.

File: mapviz_plugins/src/point_drawing_plugin.cpp (hunk 8 above)
Comment (jreyes512) [reply mapviz_plugins/src/point_drawing_plugin.cpp]: Oops, looks like I had missed one.  My bad, fixed in latest commit.

**Comment by pjreed** [comment]
This looks good now, but I'm not sure why the build is failing.  It seems like it's having an issue installing `ros-melodic-rqt-gui-cpp`, which should exist... I'll investigate.

**Pull request merged**